firefox-bootstrap/bootstrap.js

```diff
@@ -0,0 +1,224 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+let { interfaces: Ci, utils: Cu } = Components;
+
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+
+XPCOMUtils.defineLazyModuleGetter(this, "CustomizableUI",
+  "resource:///modules/CustomizableUI.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "Services",
+                                  "resource://gre/modules/Services.jsm");
+
+const PREF_WC_REPORTER_ENABLED = "extensions.webcompat-reporter.enabled";
+const WIDGET_ID = "webcompat-reporter-button";
+
+// TODO: send custom headers
+// TODO: set pref in all.js
+
+let WebCompatReporter = {
+  win: null,
+  windowListener: {
+    onOpenWindow: function(window) {
+      this.win = window.QueryInterface(Ci.nsIInterfaceRequestor). // eslint-disable-line new-cap
+        getInterface(Ci.nsIDOMWindow);
+
+      this.win.addEventListener("DOMContentLoaded", function onLoad() {
+        this.win.removeEventListener("DOMContentLoaded", onLoad, false);
+        WebCompatReporter.setUpListeners(this.win.gBrowser);
```

Comment:
should probably just use `this` here (after `bind()`ing `onLoad` to be consistent.